nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -229,7 +229,9 @@ contract StakingEscrow is Issuer {
         internal view returns (uint16)
     {
         // if the next period (after current) is confirmed
-        if (_info.confirmedPeriod1 > _currentPeriod || _info.confirmedPeriod2 > _currentPeriod) {
+        if (_info.windDown &&
```

Comment:
Just to make sure I understand here, we add 1 because the next period has already been confirmed, and the duration (`periods`) has been adjusted (`-1 if winddown, and +1 if !winddown`)...?

Reply:
very close, but not precisely.
we have several invariants with winding down parameter:
- state when whinding down is disabled. in that we case we don't need to do anything with duration and `confirmedPeriod`. Duration is fixed and we use it straithforward in last period calculation.
- winding dowin is enabled: duration is fixed without confimrations, but when worker confirms then duration must be decresed and only starting from the next period. As a result we use duration straightforward without confirmation and +1 with.

Next step is switching between invariats (when change winding down parameter). Without confirmation both invariants work equaly, so no need aditional math. 
When next period is confirmed:
- from false to true. We have "deferred" decrease of duration for the next period, so we return it back, mean add 1 to duration.
- from true to false. Same but in other direction

Also `setWindDown` has some corner cases, but logic there is the same as above